src/main/java/reactor/netty/transport/ProxyProvider.java

```diff
@@ -393,11 +499,21 @@ public ProxyProvider build() {
 		 * list of hosts that should be reached directly, bypassing the proxy.
 		 *
 		 * @param nonProxyHostsPattern Regular expression (<code>using java.util.regex</code>)
-		 * for a configured list of hosts that should be reached directly, bypassing the proxy.
+		 *                             for a configured list of hosts that should be reached directly, bypassing the proxy.
 		 * @return {@code this}
 		 */
 		Builder nonProxyHosts(String nonProxyHostsPattern);
 
+		/**
+		 * A standard predicate expression for a configured list of hosts that should be reached directly, bypassing
+		 * the proxy.
+		 *
+		 * @param nonProxyHostsPredicate A Predicate {@link Predicate} on {@link SocketAddress} to determine if the
```

Comment:
ideally the javadoc should be rephrased a bit: "to determine if address should not go through" is weird to read. maybe `that returns {@literal true} if the host should bypass the configured proxy`

Reply:
👍